src/artm/core/master_component.cc

```diff
@@ -350,7 +356,8 @@ void MasterComponent::RequestProcessBatches(const ProcessBatchesArgs& process_ba
     auto pi = std::make_shared<ProcessorInput>();
     pi->set_notifiable(&batch_manager);
     pi->set_scores_merger(scores_merger);
-    pi->set_cache_manager(cache_manager_ptr);
+    pi->set_cache_manager(theta_cache_manager_ptr);
+    pi->set_ptdw_cache_manager(ptdw_cache_manager_ptr);
     pi->set_model_name(model_name);
     pi->set_batch_filename(args.batch_filename(batch_index));
     pi->set_batch_weight(args.batch_weight(batch_index));
```

Comment:
Eventually we should be able to completely remove ``ProcessBatchesArgs.use_ptdw_matrix``. For now I think we should add the following logic:
```
if (process_batches_args.theta_matrix_type == Ptdw)
  pi->mutable_model_config()->set_use_ptdw_matrix(true)
```